cmd/account/account-list.go

```diff
@@ -0,0 +1,179 @@
+	return &accountListOptions{
+		flags:      flags,
+		printFlags: printer.NewPrintFlags(),
+		IOStreams:  streams,
+	}
+}
+
+func newCmdAccountList(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags) *cobra.Command {
+	ops := newAccountListOptions(streams, flags)
+	accountListCmd := &cobra.Command{
+		Use:               "list",
+		Short:             "List out accounts for username",
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ops.complete(cmd, args))
+			cmdutil.CheckErr(ops.run())
+		},
+	}
+	ops.printFlags.AddFlags(accountListCmd)
+	accountListCmd.Flags().StringVarP(&ops.output, "output", "o", "", "Output format. One of: json|yaml|jsonpath=...|jsonpath-file=... see jsonpath template [http://kubernetes.io/docs/user-guide/jsonpath].")
+	accountListCmd.Flags().StringVarP(&ops.username, "user", "u", "", "LDAP username")
+	accountListCmd.Flags().StringVarP(&ops.payerAccount, "payer-account", "p", "", "Payer account type")
+	accountListCmd.Flags().StringVarP(&ops.accountID, "account-id", "i", "", "Account ID")
+
+	return accountListCmd
+}
+
+func (o *accountListOptions) complete(cmd *cobra.Command, _ []string) error {
+	if o.payerAccount == "" {
+		return cmdutil.UsageErrorf(cmd, "Payer account was not provided")
+	}
+	if o.username != "" && o.accountID != "" {
+		return cmdutil.UsageErrorf(cmd, "Cannot provide both username and account ID")
+	}
+	var err error
+	o.kubeCli, err = k8s.NewClient(o.flags)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *accountListOptions) run() error {
+
+	var (
+		tempAccountIDs []string
+		user           string
+		OuID           string
+	)
+	// Instantiate Aws client
+	awsClient, err := awsprovider.NewAwsClient(o.payerAccount, "us-east-1", "")
+	if err != nil {
+		return err
+	}
+
+	if o.payerAccount == "osd-staging-2" {
+		OuID = "ou-rs3h-ry0hn2l9"
+	} else if o.payerAccount == "osd-staging-1" {
+		OuID = "ou-0wd6-z6tzkjek"
+	}
+
+	input := &organizations.ListAccountsForParentInput{
+		ParentId: &OuID,
+	}
+	accounts, err := awsClient.ListAccountsForParent(input)
+	if err != nil {
+		return err
+	}
+	// Initialize hashmap
+	m := make(map[string][]string)
+
+	if o.username == "" {
+		// Loop through list of accounts and build hashmap of users and accounts
+		for _, a := range accounts.Accounts {
+			inputListTags := &organizations.ListTagsForResourceInput{
+				ResourceId: a.Id,
+			}
+			tagVal, err := awsClient.ListTagsForResource(inputListTags)
+			if err != nil {
+				return err
+			}
+			for _, t := range tagVal.Tags {
+				if t.Key == aws.String("owner") {
+					user = *t.Value
+				}
+			}
+			// Check if user is already in the hashmap
+			// If yes append to list of accounts
+			// If no create list of accounts and add it as value to key
+			_, ok := m[user]
+			if ok {
+				_ = append(m[user], *a.Id)
```

Comment:
Why use two practices for array expansion in the same fnction?


Which is the one you would like to follow repo wide and choose it

Reply:
I'm not sure I follow what you're asking in this comment.  Specifically what's happening here is that we're checking the key exists in the map before attempting to append the ID to it. Otherwise we'll get panics.  If the key doesn't exist, we initialize a new list as the value for that key.